browser/resources/settings/brave_sync_page/brave_sync_subpage.ts

```diff
@@ -100,58 +100,76 @@ class SettingBraveSyncSubpage extends SettingBraveSyncSubpageBase {
 
-  /** @private {?SyncBrowserProxy} */
-  browserProxy_ = SyncBrowserProxyImpl.getInstance()
+  private browserProxy_ = SyncBrowserProxyImpl.getInstance()
 
-  braveSyncBrowserProxy_ = BraveSyncBrowserProxy.getInstance();
+  private braveSyncBrowserProxy_: BraveSyncBrowserProxy =
+    BraveSyncBrowserProxy.getInstance()
 
   /**
-  * The beforeunload callback is used to show the 'Leave site' dialog. This
-  * makes sure that the user has the chance to go back and confirm the sync
-  * opt-in before leaving.
-  *
-  * This property is non-null if the user is currently navigated on the sync
-  * settings route.
-  *
-  * @private {?Function}
-  */
-  beforeunloadCallback_ = null
+   * The beforeunload callback is used to show the 'Leave site' dialog. This
+   * makes sure that the user has the chance to go back and confirm the sync
+   * opt-in before leaving.
+   *
+   * This property is non-null if the user is currently navigated on the sync
+   * settings route.
+   */
+  private beforeunloadCallback_: ((e: Event) => void) | null = null
 
   /**
-  * The unload callback is used to cancel the sync setup when the user hits
-  * the browser back button after arriving on the page.
-  * Note: Cases like closing the tab or reloading don't need to be handled,
-  * because they are already caught in |PeopleHandler::~PeopleHandler|
-  * from the C++ code.
-  *
-  * @private {?Function}
-  */
-  unloadCallback_ = null
+   * The unload callback is used to cancel the sync setup when the user hits
+   * the browser back button after arriving on the page.
+   * Note: Cases like closing the tab or reloading don't need to be handled,
+   * because they are already caught in |PeopleHandler::~PeopleHandler|
+   * from the C++ code.
+   */
+  private unloadCallback_: ((e: Event) => void) | null = null
 
   /**
-  * Whether the user completed setup successfully.
-  * @private {boolean}
-  */
-  setupSuccessful_ = false
+   * Whether the user completed setup successfully.
+   */
+  private setupSuccessful_: boolean = false
 
-  attached() {
-    super.attached()
+  static get observers() {
+    return [
+      'updatePageStatus_(syncStatus.*)'
+    ]
+  }
+
+  override currentRouteChanged() {
     const router = Router.getInstance()
-    if (router.getCurrentRoute() == router.getRoutes().BRAVE_SYNC_SETUP) {
+    if (router.getCurrentRoute() === router.getRoutes().BRAVE_SYNC_SETUP) {
       this.onNavigateToPage_()
+      return
     }
+
+    if (router.getRoutes().BRAVE_SYNC_SETUP.
+      contains(router.getCurrentRoute())) {
+      return
+    }
+
+    this.onNavigateAwayFromPage_()
   }
 
-  /** @override */
-  detached() {
+  override connectedCallback() {
+    super.connectedCallback()
     const router = Router.getInstance()
-    if (router.getRoutes().BRAVE_SYNC_SETUP.contains(router.getCurrentRoute())) {
+    if (router.getCurrentRoute() === router.getRoutes().BRAVE_SYNC_SETUP) {
+      this.onNavigateToPage_()
+    }
```

Comment:
@AlexeyBarabash I think the linter recommended making this simplification, but not sure if it's correct.

Reply:
I think it is - `getCurrentRoute` returns an object, which will be doing strict equality anyway. The only place this might bite us is if we use `null/undefined` interchangably.